modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/lowstorage/LowStorageDialog.kt

```diff
@@ -0,0 +1,169 @@
+import au.com.shiftyjelly.pocketcasts.compose.buttons.RowButton
+import au.com.shiftyjelly.pocketcasts.compose.buttons.RowOutlinedButton
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH20
+import au.com.shiftyjelly.pocketcasts.compose.components.TextH50
+import au.com.shiftyjelly.pocketcasts.compose.preview.ThemePreviewParameterProvider
+import au.com.shiftyjelly.pocketcasts.compose.theme
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import au.com.shiftyjelly.pocketcasts.utils.Util
+import kotlinx.coroutines.launch
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+
+@Composable
+fun LowStorageLaunchBottomSheet(
+    parent: ViewGroup,
+    modifier: Modifier = Modifier,
+    shouldShow: Boolean = true,
+    totalDownloadSize: Long,
+    onManageDownloadsClick: () -> Unit,
+    onMaybeLaterClick: () -> Unit,
+    onExpanded: () -> Unit,
+) {
+    val sheetState = rememberModalBottomSheetState(
+        initialValue = ModalBottomSheetValue.Hidden,
+        skipHalfExpanded = true,
+    )
+    val coroutineScope = rememberCoroutineScope()
+
+    ModalBottomSheet(
+        parent = parent,
+        shouldShow = shouldShow,
+        customSheetState = sheetState,
+        onExpanded = onExpanded,
+        customContent = {
+            LowStorageDialog(
+                modifier = modifier,
+                totalDownloadSize = totalDownloadSize,
+                onManageDownloadsClick = {
+                    coroutineScope.launch { sheetState.hide() }
+                    onManageDownloadsClick.invoke()
+                },
+                onMaybeLaterClick = {
+                    coroutineScope.launch { sheetState.hide() }
+                    onMaybeLaterClick.invoke()
+                },
+            )
+        },
+    )
+}
+
+@Composable
+internal fun LowStorageDialog(
+    modifier: Modifier = Modifier,
+    totalDownloadSize: Long,
+    onManageDownloadsClick: () -> Unit,
+    onMaybeLaterClick: () -> Unit,
+) {
+    val formattedTotalDownloadSize = Util.formattedBytes(bytes = totalDownloadSize, context = LocalContext.current)
+
+    Box(
+        modifier = modifier
+            .fillMaxWidth(),
+        contentAlignment = Alignment.Center,
+    ) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = modifier.fillMaxWidth(),
+        ) {
+            Image(
+                painterResource(IR.drawable.swipe_affordance),
+                contentDescription = stringResource(LR.string.swipe_affordance_content_description),
+                modifier = modifier
+                    .width(56.dp)
+                    .padding(top = 8.dp, bottom = 24.dp),
+            )
+
+            Icon(
+                painterResource(IR.drawable.pencil_cleanup),
+                contentDescription = stringResource(LR.string.pencil_clean_up_icon_content_description),
+                modifier = modifier
```

Comment:
```suggestion
                modifier = Modifier
```
Only the root component should use the modifier passed into the method.